evm-template/runtime/src/configs/xcm_config.rs

```diff
@@ -372,3 +390,202 @@ parameter_types! {
+        match currency {
+            CurrencyId::SelfReserve => {
+                let multi: Location = SelfReserve::get();
+                Some(multi)
+            }
+            CurrencyId::ForeignAsset(asset) => AssetXConverter::convert_back(&asset),
+            CurrencyId::Erc20 { contract_address } => {
+                let mut location = Erc20XcmBridgePalletLocation::get(); // TODO: if we do not have an alternative, we need to import this pallet from moonbeam
+                location
+                    .push_interior(Junction::AccountKey20 {
+                        key: contract_address.0,
+                        network: None,
+                    })
+                    .ok();
+                Some(location)
+            }
+        }
+    }
+}
+
+/// Wrapper type around `LocationToAccountId` to convert an `AccountId` to type `H160`.
+pub struct LocationToH160;
+impl ConvertLocation<H160> for LocationToH160 {
+    fn convert_location(location: &Location) -> Option<H160> {
+        <LocationToAccountId as ConvertLocation<AccountId>>::convert_location(location)
+            .map(Into::into)
+    }
+}
+
+parameter_types! {
+    // This is the relative view of erc20 assets.
+    // Identified by this prefix + AccountKey20(contractAddress)
+    // We use the RELATIVE multilocation
+    pub Erc20XcmBridgePalletLocation: Location = Location {
+        parents:0,
+        interior: [
+            PalletInstance(<Erc20XcmBridge as PalletInfoAccess>::index() as u8)
+        ].into()
+    };
+
+    // To be able to support almost all erc20 implementations,
+    // we provide a sufficiently high gas limit.
+    pub Erc20XcmBridgeTransferGasLimit: u64 = 800_000;
+}
+
+impl pallet_erc20_xcm_bridge::Config for Runtime {
+    type AccountIdConverter = LocationToH160;
+    type Erc20MultilocationPrefix = Erc20XcmBridgePalletLocation;
+    type Erc20TransferGasLimit = Erc20XcmBridgeTransferGasLimit;
+    type EvmRunner = pallet_evm::runner::stack::Runner<Self>;
+}
+
+/// Provider for the DOT asset reserve.
+pub struct DOTReserveProvider;
+
+impl Reserve for DOTReserveProvider {
+    fn reserve(asset: &Asset) -> Option<Location> {
+        let AssetId(location) = &asset.id;
+
+        // If the location is `{ parents: 1, location: Here }`
+        let dot_here = Location::new(1, Here);
+        let dot_asset_hub = AssetHubLocation::get();
+
+        if location == &dot_here || location == &dot_asset_hub {
+            Some(dot_asset_hub) // Reserve is on AssetHub.
+        } else {
+            None
+        }
+    }
+}
+
+/// Provider for bridged assets' reserve.
+pub struct BridgedAssetReserveProvider;
+
+impl Reserve for BridgedAssetReserveProvider {
+    fn reserve(asset: &Asset) -> Option<Location> {
+        let AssetId(location) = &asset.id;
+
+        let asset_hub_reserve = AssetHubLocation::get();
+
+        match location.interior() {
+            X1(arc) if arc.as_ref() == &[GlobalConsensus(Ethereum { chain_id: 1 })] =>
```

Comment:
I would make this check looser. So any asset that has `parents > 1` and `interior` that starts with `GlobalConsensus(_)` pattern can be considered a bridged asset.